scipy/signal/tests/test_signaltools.py

```diff
@@ -3912,117 +3991,165 @@ def test_nonnumeric_dtypes(func, xp):
         func(*args, x=1.)
 
 
-@skip_xp_backends(np_only=True)
-@pytest.mark.parametrize('dt', 'fdFD')
+@skip_xp_backends(cpu_only=True)
+@pytest.mark.parametrize('dt', ['float32', 'float64', 'complex64', 'complex128'])
 class TestSOSFilt:
 
     # The test_rank* tests are pulled from _TestLinearFilter
+    @skip_xp_backends('jax.numpy', reason='buffer array is read-only')
     def test_rank1(self, dt, xp):
-        x = np.linspace(0, 5, 6).astype(dt)
-        b = np.array([1, -1]).astype(dt)
-        a = np.array([0.5, -0.5]).astype(dt)
+        dt = getattr(xp, dt)
+        x = xp.linspace(0, 5, 6, dtype=dt)
+        b = xp.asarray([1, -1], dtype=dt)
+        a = xp.asarray([0.5, -0.5], dtype=dt)
 
         # Test simple IIR
-        y_r = np.array([0, 2, 4, 6, 8, 10.]).astype(dt)
+        y_r = xp.asarray([0, 2, 4, 6, 8, 10.], dtype=dt)
         sos = tf2sos(b, a)
-        assert_array_almost_equal(sosfilt(tf2sos(b, a), x), y_r)
+        sos = xp.asarray(sos)   # XXX until tf2sos is numpy only
```

Comment:
did you mean this? If so, the same appears below
```suggestion
        sos = xp.asarray(sos)   # XXX while tf2sos is numpy only
```

Reply:
Oh, thanks! Apparently I changed the wording mid-phrase. Fixed in the last commit.